### ISSUE TITLE
How does the plugin handle multiple users sharing the same database?

### ISSUE THREAD

**Issue by ams-tschoening** [marvinweber/KPSimpleBackup#32]
This is a follow-up to #31: One SAMBA-share hosting a database used by multiple users. All of those use KPSimpleBackup to create additional backups when each one saves its own changes.

Is that a problem when NOT having concurrent changes? So really only one user after another opens, changes and saves? Looking at file names, I assume KPSimpleBackup most likely won't care who created a backup and simply clean things up as everything has been created by one and the same user?

What happens in case of concurrent access? E.g when user 1 opened the file, changed something, user 2 did the same in the meantime and saved before user 1 could. Regarding my tests, KeePass would recognize that and provide options to either synchronize or overwrite the database, in which case a user would need to synchronize most likely.

But when would come KPSimpleBackup into play in this scenario? First when user 2 is saving of course, but what about user 1? Would it create backups when KeePass detected concurrent access already or only after user 1 would have decided how to deal with it and the result really gets saved again?

**Comment by marvinweber** [comment]
#### No Concurrent Changes
This should not be a problem. The plugin simply exports/saves the database again according to the date-time format you defined in the settings of the plugin. The cleanup deletes old files based on their creation date, so even different namings (i.e. different date-time formats) shouldn't be a probleme here.

#### Concurrent Changes
Assume two users A and B. Both open the database an make changes, A saves first, then B.  
As A can save as usual backups will also be created as usual.  
If B saves now the "Someone changed the file" popup appears. The Backup takes only place after KeePass has either finished overwriting or synchronizing/merging both files into one new file.

#### TL;DR
KPSimpleBackup creates a backup of whatever is saved by KeePass at the end.


---

Does this answer your questions? :)
